### PR TITLE
Ticket 861 csv

### PR DESCRIPTION
**Pull request by paulwellsagilej**

### CODE REVIEW

**Comment by steve-daly** [comment]
Hi @paulwellsagilej this is title "Ticket 861 csv" but seems to include other commits such as the Windows Batch File changes.  Was that intentional?

**Comment by steve-daly** [comment]
Can I also check about the CSV column heading implementation. In Issue #861 the spec suggested that repeated column heading should have numeric suffixes (e.g. PUID2 as in the example given).  Excel needs unique column headings for some cases (e.g. Pivot Tables, or index lookups) so I think we'd want to avoid having duplicated column headings.

**Comment by paulwellsagilej** [comment]
@steve-daly According to my local comparison between branches I have changed only two files, so I'm investigating why GitHub thinks nearly 1400 files have changed. Probably a fight over line endings as suggested by @sparkhi. Thanks for the nudge about keeping headers unique - I have just pushed a change for that.

**Comment by sparkhi** [comment]
This PR should not be merged yet. There is an issue around bean instantiation using spring, so the changes in this PR would succeed the build but not launch droid.

File: droid-results/src/main/java/uk/gov/nationalarchives/droid/profile/CsvItemWriter.java [hunk 1]
```diff
@@ -29,354 +31,365 @@
...
-
-    private static String toFileName(String name) {
-        return FilenameUtils.getName(name);
-    }
-    
-    /**
-     * No config is needed by this class, but it's retained temporarily for backwards compatibility purposes.
-     * @param config the config to set
-     */
-    public void setConfig(DroidGlobalConfig config) {
```
Comment (sparkhi) [inline droid-results/src/main/java/uk/gov/nationalarchives/droid/profile/CsvItemWriter.java:268]: Removal of this setter causes problem in spring config

**Review by sparkhi — COMMENTED** [review]
There are a few problems that need to be addressed before this PR can be merged 
1) Removal of the setter method for config, makes it fail at runtime as the spring config tries to call this setter method
2) Folders do not have format count and consequently there is a NPE thrown on a folder as we compare for column count
3) When we run the export again, the header for "HASH" is removed but the data is retained which is a critical problem.

**Comment by sparkhi** [comment]
The work is done through other branch

**Pull request closed without merging**